frontends/p4/p4-parse.ypp

```diff
@@ -813,6 +821,24 @@ actionRef
                                            $$ = new IR::ActionListElement(@2, $1, mce); }
     ;
 
+entry
+    : optAnnotations keysetExpression ':' actionRef
+                                         { if($2->is<IR::ListExpression>())
```

Comment:
I think you should follow the handling of SelectCase, and allow here just an expression.
default in particular is special; (default) is not the same as default.